plugin/src/main/java/org/opensearch/ml/utils/MLNodeUtils.java

```diff
@@ -89,6 +90,39 @@ public static void validateSchema(String schemaString, String instanceString) th
         }
     }
 
+    /**
+     * This method processes the input JSON string and replaces the string values of the parameters with JSON objects if the string is a valid JSON.
+     * @param inputJson The input JSON string
+     * @return The processed JSON string
+     */
+    public static String processRemoteInferenceInputDataSetParametersValue(String inputJson) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode rootNode = mapper.readTree(inputJson);
+
+        if (rootNode.has("parameters") && rootNode.get("parameters").isObject()) {
+            ObjectNode parametersNode = (ObjectNode) rootNode.get("parameters");
+
+            parametersNode.fields().forEachRemaining(entry -> {
+                String key = entry.getKey();
+                JsonNode value = entry.getValue();
+
+                if (value.isTextual()) {
+                    String textValue = value.asText();
+                    try {
+                        // Try to parse the string as JSON
+                        JsonNode parsedValue = mapper.readTree(textValue);
```

Comment:
can it handle this case that a parameter is string, but it's a json string. For example 
`"content": "{\"title\": \"hello\"}"`